tests/Exceptions/ApiExceptionTest.php

```diff
@@ -59,4 +59,23 @@ public function testExceptionWithNoHttpBody(): void
             $this->assertEquals($expectedExceptionToString, (string) $apiException);
         }
     }
+
+    public function testRethrowWithHintException(): void
+    {
+        try {
+            $e = new \Exception('Any error message that caused the root problem should be shown');
+
+            throw ApiException::rethrowWithHint($e, 'deleteDocuments');
+        } catch (\Exception $apiException) {
```

Comment:
`expectException` should be used instead of catching

Reply:
Actually, this is impossible because I need to assert if the previous exception error message is preserved. Unless you know how to do it.

Also, your commend made me realize I don't need to throw, I could just ensure the messages are how I want.